app/src/main/java/com/duckduckgo/app/fire/DataClearingStore.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.fire
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.support.annotation.VisibleForTesting
+import androidx.core.content.edit
+import javax.inject.Inject
+
+
+interface DataClearingStore {
```

Comment:
This seems pretty generic, but is only used by the 🔥 pixel.  If someone were to come along and think they can reuse this for their pixels without knowing the history they would introduce some bugs.

A couple of potential options:
1) This should probably be clearer about its intent `incrementUnsentFirePixelCount` (which doesn't seem very nice, but is probably the simplest)

2) Make it more generic in its interface by taking the name of the pixel to count (but this is about to get extra complicated since some pixels can now take parameters)

Reply:
What about if we rename the interface to `DataClearingUnsentPixelStore`? Or `UnsentDataClearingPixelStore`?

Reply:
I'm less worried about the interface name, though having `Pixel` in there would probably be helpful.  It really depends if you want to keep this thing specific to the "Forget All" functionality or keep it named generically, but make it function generically as well.

I would probably go with `UnsentForgetAllPixelStore` as the interface name and leave the function names as is, rather than implementing generic functionality which YAGNI :)